tests/test_client.py

```diff
@@ -0,0 +1,184 @@
+    assert "is enabled for this system" in enable
+
+
+def test_install_katello_host_tools(client, foreman_version):
+    release = client.system_info.release.split('.')[0]
+    client.run_expect([0], "yum -y install http://yum.theforeman.org/client/%s/el%s/x86_64/foreman-client-release.rpm" % (foreman_version, release))
+    client.run_expect([0], 'yum -y install katello-host-tools')
+    katello_host_tools = client.package('katello-host-tools')
+    assert katello_host_tools.is_installed
+
+
+def test_install_package_local(client):
+    client.run('yum -y install walrus-0.71')
+    walrus = client.package('walrus')
+    assert walrus.is_installed
+    assert walrus.version == '0.71'
+
+
+def test_available_errata(client, smoker, entities):
+    time.sleep(10)
+    hostname = client.check_output('hostname -s')
+    errata = smoker.check_output("hammer host errata list --host %s" % hostname)
+    assert entities['errata_id'] in errata
+
+
+@pytest.mark.katello_agent
+def test_install_katello_agent(client):
+    release = client.system_info.release.split('.')[0]
+
+    if release != 6 or release != 7:
+        pytest.skip("katello-agent not available on EL8")
+
+    client.run_expect([0], 'yum -y install katello-agent')
+    katello_agent = client.package('katello-agent')
+    assert katello_agent.is_installed
+
+
+@pytest.mark.katello_agent
+def test_katello_agent_package_install(client, smoker):
+    release = client.system_info.release.split('.')[0]
+
+    if release != 6 or release != 7:
+        pytest.skip("katello-agent not available on EL8")
+
+    hostname = client.check_output('hostname -s')
+    smoker.run("timeout 300 hammer host package install --host %s --packages gorilla" % hostname)
+    gorilla = client.package('gorilla')
+    assert gorilla.is_installed
+
+
+@pytest.mark.katello_agent
+def test_katello_agent_errata_install(client, smoker, entities):
+    release = client.system_info.release.split('.')[0]
+
+    if release != 6 or release != 7:
+        pytest.skip("katello-agent not available on EL8")
+
+    hostname = client.check_output('hostname -s')
+    smoker.run("timeout 300 hammer host errata apply --errata-ids '%s' --host %s" % (entities['errata_id'], hostname))
+    walrus = client.package('walrus')
+    assert walrus.is_installed
+    assert walrus.version == '5.21'
+
+
+@pytest.mark.katello_agent
+def test_katello_agent_remove_package(client, smoker):
+    release = client.system_info.release.split('.')[0]
+
+    if release != 6 or release != 7:
+        pytest.skip("katello-agent not available on EL8")
+
+    hostname = client.check_output('hostname -s')
+    client.run("timeout 300 hammer host package remove --host %s --packages walrus" % hostname)
+
+    walrus = client.package('walrus')
+    assert not walrus.is_installed
+
+
+@pytest.mark.container
+def test_fetch_container_content(client, smoker, entities, server_hostname):
```

Comment:
This test is kind of ugly. As you will note I am currently testing this on smoker not the client. Running podman inside podman is difficult without using a `--privileged` container.

Reply:
How about [skopeo](https://github.com/containers/skopeo):
```console
# mkdir containers
# skopeo copy docker://registry.fedoraproject.org/fedora:latest dir:containers
Getting image source signatures
Copying blob 8fde7942e775 done
Copying config 79fd58dc76 done
Writing manifest to image destination
Storing signatures
# ls containers/
79fd58dc76113dac76a120f22cadecc3b2d1794b414f90ea368cf66096700053  8fde7942e775327f2d6ddaf08f45fda74201a7769951b0413860e21b59b4bf82  manifest.json  version
```

Also has an option to log in. There are also options to inspect them from a remote source so you don't need to download it all.

Reply:
That suggestion worked.